azurelinuxagent/common/future.py

```diff
@@ -137,7 +137,7 @@ def is_file_not_found_error(exception):
 
     if sys.version_info[0] == 2:
         # Python 2 uses OSError(errno=2)
-        return isinstance(exception, OSError) and exception.errno == 2
+        return isinstance(exception, (IOError, OSError)) and exception.errno == 2
```

Comment:
Some weirdness here; `IOError` was originally used, but that fails on our end-to-end testing for, e.g., centos 7.8 images. `OSError` seemed to fix this failure, until the change in `default.py`, using this function.

The python documentation mentions that the error should be `IOError` for python 2, but we need both here.

Reply:
could you add a unit test that illustrates this?